### PR TITLE
Refactored smart contract queries

### PR DESCRIPTION
**Pull request by popenta**
Removed the old interfaces and adapter and replaced with the new DTOs.

### CODE REVIEW

File: multiversx_sdk/controllers/smart_contract_controller.py [hunk 1]
```diff
@@ -21,35 +22,8 @@
...
-    return_message: str
-    gas_used: int
-
-    def get_return_data_parts(self) -> List[bytes]:
-        ...
-
-
 class INetworkProvider(Protocol):
-    def query_contract(self, query: IQuery) -> IQueryResponse:
+    def query_contract(self, query: SmartContractQuery) -> SmartContractQueryResponse:
```
Comment (andreibancioiu) [inline multiversx_sdk/controllers/smart_contract_controller.py:26]: :+1: Indeed, it's reasonable & practical to have DTOs as input & output of functions in interfaces.

File: multiversx_sdk/network_providers/proxy_network_provider.py [hunk 2]
```diff
@@ -178,10 +179,22 @@ def await_transaction_on_condition(self,
...
+    def query_contract(self, query: SmartContractQuery) -> SmartContractQueryResponse:
+        query_converter = SmartContractQueryConverter()
+        request = query_converter.smart_contract_query_to_dictionary(query)
         response = self.do_post_generic('vm-values/query', request)
-        return ContractQueryResponse.from_http_response(response.get('data', ''))
+        response = response.get('data', '')
+
+        return_data = response.get('returnData', []) or response.get('ReturnData', [])
+        return_code = response.get('returnCode', '') or response.get('ReturnCode', '')
+        return_message = response.get('returnMessage', '') or response.get('ReturnMessage', '')
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/proxy_network_provider.py:190]: Now I see. Fine for now, let's think of other ways to tackle this (in the background) :+1: 

Our converter for `SmartContractQuery` has knowledge of protocol-desired JSON-ifying (which seems fine in practice, although it somehow promises that it converts the query to a dictionary, not necessarily to a protocol-desired dictionary).

On the other hand, here we have a way of converting the bare HTTP response payload to our DTO, the `SmartContractQueryResponse`.

Thus, there's a design asymmetry. 

Can stay as it is for now.

File: multiversx_sdk/network_providers/api_network_provider.py [hunk 3]
```diff
@@ -125,10 +124,21 @@ def get_non_fungible_token(self, collection: str, nonce: int) -> NonFungibleToke
...
-        request = ContractQueryRequest(query).to_http_request()
+    def query_contract(self, query: SmartContractQuery) -> SmartContractQueryResponse:
+        query_converter = SmartContractQueryConverter()
+        request = query_converter.smart_contract_query_to_dictionary(query)
         response = self.do_post_generic('query', request)
-        return ContractQueryResponse.from_http_response(response)
+
+        return_data = response.get('returnData', []) or response.get('ReturnData', [])
+        return_code = response.get('returnCode', '') or response.get('ReturnCode', '')
+        return_message = response.get('returnMessage', '') or response.get('ReturnMessage', '')
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/api_network_provider.py:134]: Somehow duplication, but can be kept as it is for now (until we tackle the design asymmetry mentioned in the other comment).

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk/network_providers/api_network_provider.py (hunk 3 above)
Comment (andreibancioiu) [reply multiversx_sdk/network_providers/api_network_provider.py:134]: As discusses, transformations and conversions can be done in `shared.py` or something similar (not in the converters package).

File: multiversx_sdk/network_providers/proxy_network_provider.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk/network_providers/proxy_network_provider.py:190]: As discussed, will do this in the next PR.

**Review by danielailie — APPROVED** [review]

**Pull request merged**